### PR TITLE
move block verification to sealEngineFace

### PR DESCRIPTION
**Pull request by winsvega**
make verification possible on NoProof engine

### CODE REVIEW

File: libethcore/SealEngine.cpp [hunk 1]
```diff
@@ -31,7 +31,57 @@ void NoProof::init()
 
 void SealEngineFace::verify(Strictness _s, BlockHeader const& _bi, BlockHeader const& _parent, bytesConstRef _block) const
 {
-	_bi.verify(_s, _parent, _block);
+    _bi.verify(_s, _parent, _block);
+    if (_parent)
```
Comment (gumb0) [inline libethcore/SealEngine.cpp]: Let's move this after `if (_s != CheckNothingNew)` as it was before

File: libethcore/SealEngine.cpp [hunk 2]
```diff
@@ -31,7 +31,57 @@ void NoProof::init()
...
+
+        if (_bi.gasLimit() < chainParams().minGasLimit)
+            BOOST_THROW_EXCEPTION(InvalidGasLimit() << RequirementError(
+                                      bigint(chainParams().minGasLimit), bigint(_bi.gasLimit())));
+
+        if (_bi.gasLimit() > chainParams().maxGasLimit)
+            BOOST_THROW_EXCEPTION(InvalidGasLimit() << RequirementError(
+                                      bigint(chainParams().maxGasLimit), bigint(_bi.gasLimit())));
+
+        if (_bi.number())
```
Comment (gumb0) [inline libethcore/SealEngine.cpp]: `if (_bi.number() && _bi.extraData().size() > chainParams().maximumExtraDataSize)` is more readable I think than nested `if`

**Comment by gumb0** [comment]
Solidity tests use NoProof, so check how it works and submit a PR to https://github.com/ethereum/solidity/blob/develop/test/RPCSession.cpp if change in config needed

**Comment by winsvega** [comment]
soltest fix 
https://github.com/ethereum/solidity/pull/4389

**Pull request merged**